zulip/integrations/bridge_with_slack/run-slack-bridge

```diff
@@ -0,0 +1,155 @@
+        # zulip-specific
+        self.zulip_client = zulip.Client(
+            email=self.zulip_config["email"],
+            api_key=self.zulip_config["api_key"],
+            site=self.zulip_config["site"])
+        self.zulip_stream = self.zulip_config["stream"]
+        self.zulip_subject = self.zulip_config["topic"]
+
+        # slack-specific
+        self.channel = self.slack_config["channel"]
+        self.slack_client = SlackClient(self.slack_config["token"])
+
+    def wrap_slack_mention_with_bracket(self, zulip_msg):
+        # type: (Dict[str, Any]) -> None
+        words = zulip_msg["content"].split(' ')
+        for w in words:
+            if w.startswith('@'):
+                zulip_msg["content"] = zulip_msg["content"].replace(w, '<' + w + '>')
+
+    def replace_slack_id_with_name(self, msg):
+        # type: (Dict[str, Any]) -> None
+        words = msg['text'].split(' ')
+        for w in words:
+            if w.startswith('<@') and w.endswith('>'):
+                _id = w[2:-1]
+                msg['text'] = msg['text'].replace(_id, self.slack_id_to_name[_id])
+
+    def zulip_to_slack(self):
+        # type: () -> Callable[[Dict[str, Any]], None]
+        def _zulip_to_slack(msg):
+            # type: (Dict[str, Any]) -> None
+            message_valid = check_zulip_message_validity(msg, self.zulip_config)
+            if message_valid:
+                self.wrap_slack_mention_with_bracket(msg)
+                slack_text = SLACK_MESSAGE_TEMPLATE.format(username=msg["sender_full_name"],
+                                                           message=msg["content"])
+                self.slack_client.rtm_send_message(
+                    self.channel,
+                    slack_text,
+                )
+        return _zulip_to_slack
+
+    def slack_to_zulip(self, msg):
+        # type: (Dict[str, Any]) -> None
+        if msg['type'] != 'message':
+            return
+        if msg['channel'] == self.channel:
+            user_id = msg['user']
+            user = self.slack_id_to_name[user_id]
+            not_from_bot = user != self.slack_config['username']
+            if not_from_bot:
+                self.replace_slack_id_with_name(msg)
+                content = ZULIP_MESSAGE_TEMPLATE.format(username=user, message=msg['text'])
+                msg_data = dict(
+                    sender=self.zulip_client.email,
+                    type="stream",
+                    to=self.zulip_stream,
+                    subject=self.zulip_subject,
+                    content=content)
+                self.zulip_client.send_message(msg_data)
+
+    def run_slack_listener(self):
+        # type: () -> None
+        interval = 1
+        sc = self.slack_client
+        ul_resp = sc.api_call("users.list")
+        if ul_resp["ok"]:
+            # See also https://api.slack.com/changelog/2017-09-the-one-about-usernames
+            self.slack_id_to_name = {u["id"]: u["profile"]["display_name"] for u in ul_resp["members"]}
```

Comment:
`user['profile']['display_name']` doesn't seem to be always set. In case it is empty, may be we should look at `user['name']` or `user['real_name']` ?

Reply:
Fixed.